DOC-5711 - Write Scaling Enhancements

jamesnocentini opened the pull request:
https://issues.couchbase.com/browse/DOC-5711

jamesnocentini committed:
https://issues.couchbase.com/browse/DOC-5711

- Initial contribution

jamiltz commented:
@bbrks , @sarathkumarsivan , @adamcfraser 

At some point before the release, this PR will need with a review for accuracy from Engineering. Only one review should be sufficient. I have assigned the corresponding JIRA ticket (in the comment above) to "The One".

jamesnocentini committed:
https://issues.couchbase.com/browse/DOC-5711

- remove note about single config with `import_docs: true`

jamesnocentini committed:
Changes:

- Add workload isolation section
- Add workload isolation diagram

jamesnocentini committed:
update whats new link to mobile-web data sync page

ibsoln committed:
Amended sharded accross text to shared across (#120)

jamesnocentini committed:
- Add Import Processing section after How to enable it
- Move Workload Isolation section above Metadata Purge Interval
- Move Reference section to the end of the page

jamiltz commented:
Needs review from @rajagp

ibsoln commented on modules/ROOT/assets/attachments/sg.yaml:
```diff
@@ -144,8 +144,24 @@ properties:
-              `enable_shared_bucket_access` needs to be `true` on all nodes participating in such a configuration where as `import_docs` can only be set to `true` on a single node.
+              Starting in Sync Gateway 2.7, all Sync Gateway nodes can be configured as import nodes. This results in performance benefits as the import process is shared across all Sync Gateway nodes.
+
+              Prior to version 2.7, `import_docs` can only be set to `true` on a single node.
```
@priya Is this still true at 2.7? If enable_shared_bucket_access = true doesn't import_docs default to 'true'? Also won't multiple nodes need it set to true in order enable the position of multiple nodes handling the import processing work load? Setting it false enables workload isolation for those nodes excluded from import processing and dedicated to handling remote clients.

ibsoln committed:
Merge branch 'release/2.7' into doc-5711-write-scaling-enhancements

ibsoln commented:
@RichardSmedley / @rajagp can you please mark as reviewed so I can complete the merge?

ibsoln committed:
Merge branch 'release/2.7' into doc-5711-write-scaling-enhancements

rajagp commented on modules/ROOT/pages/index.adoc:
```diff
@@ -2,12 +2,13 @@
-xref:deployment.adoc#channel-and-revision-cache[Read more]
+xref:shared-bucket-access.adoc[Read more]
 
 == Release Notes
```
Please include following 
**Write Throughput Enhancements** 
We have improved the per-node throughput while processing writes that come in from Couchbase Lite clients [replicating](https://docs.couchbase.com/couchbase-lite/2.7/swift.html#starting-a-replication) with the Sync Gateway, as well as writes that are made via the Sync Gateway [REST API](https://docs.couchbase.com/sync-gateway/current/rest-api.html#/document).

**Import Processing Throughput** 
we have also significantly improved the performance of writes coming in from the Couchbase Server that are [imported](https://docs.couchbase.com/sync-gateway/2.6/config-properties.html#databases-foo_db-enable_shared_bucket_access) by the Sync Gateway . In addition to increasing the per-node import throughput, the enhancements also enable to scale linearly with the number of nodes. 

**High Availability of Imports**
In 2.7, import processing is automatically sharded across all the Sync Gateway nodes in a cluster. That way, even if the import node goes down, other nodes in the cluster can pick up the import shard. That way, import processing can continue uninterrupted with the other nodes in the cluster.

rajagp replied:
@ibsoln  Please update the Whats New Page . You will need to add links
